sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -284,9 +286,17 @@ object JdbcUtils extends Logging {
     val rddSchema = df.schema
     val getConnection: () => Connection = createConnectionFactory(url, properties)
     val batchSize = properties.getProperty(JDBC_BATCH_INSERT_SIZE, "1000").toInt
-    df.foreachPartition { iterator =>
-      savePartition(getConnection, table, iterator, rddSchema, nullTypes, batchSize, dialect)
-    }
+    val isolationLevel =
+      properties.getProperty(JDBC_TXN_ISOLATION_LEVEL, "READ_UNCOMMITTED") match {
```

Comment:
Maybe here, we can use a best-effort heuristics. If the `supportsTransactionIsolationLevel` calls for all the isolation levels return `false` (ideally, we should not hit this scenario), we can still keep the existing approach (i.e., do not set the isolation level).